app.go

```diff
@@ -81,9 +83,21 @@ func (f optionFunc) apply(app *App) { f(app) }
 //
 // See the documentation for go.uber.org/dig for further details.
 func Provide(constructors ...interface{}) Option {
-	return optionFunc(func(app *App) {
-		app.provides = append(app.provides, constructors...)
-	})
+	return provideOption(constructors)
+}
+
+type provideOption []interface{}
+
+func (po provideOption) apply(app *App) {
+	app.provides = append(app.provides, po...)
+}
+
+func (po provideOption) String() string {
+	items := make([]string, len(po))
+	for i, c := range po {
+		items[i] = fxreflect.FuncName(c)
+	}
+	return fmt.Sprintf("fx.Provide(%v)", strings.Join(items, ", "))
```

Comment:
nit: can use `%s` here, since we know it's a string. Same in the other `String` methods.